File: JanEasy2021/MEX.cpp
/*Result: 10/100 TLE
Naive solution in C++
Same approach as my Python solution 
Sometimes HackerEarth gives more points to C++ solutions with similar Time Complexities as Python solutions
*/
#include <bits/stdc++.h>
using namespace std;
 
#define lli long long int
#define str string
#define MOD 1000000007
#define pb push_back
#define timesaver ios_base::sync_with_stdio(false); cin.tie(NULL)
int main()
{
    timesaver;
    
    int t;
    cin >> t;
    while(t--) {
        int n;
        cin >> n;
        int arr[n];
 
        for(int i = 0; i < n; ++i) {
            cin >> arr[i];
        }
 
        set<int> divs;
 
        for(int i = 0; i < n; ++i) {
            for(int j = i+1; j < n; ++j) {
                divs.insert(__gcd(arr[i], arr[j]));
            }
        }
 
        int test = 1;
 
        while(1) {
            if(divs.find(test) == divs.end()) {
                cout << test << '\n';
                break;
            }
            ++test;
        }
    }
    
    return 0;
}

File: JanEasy2021/MEX.py
#Result: 10/100 TLE
#Solved: 0:30:00
'''
    Naive solution
    Find all gcds of every subarray of size 2 -> Time Complexity: O(n^2)
    Find first natural number not in resulting set -> Time Complexity: O(100000) (max gcd possible is 100000)

'''
from math import gcd
for tc in range(int(input())):
  n = int(input())
  arr = [int(x) for x in input().split()]
  new = set()
  arr = list(set(arr))
  n = len(arr)
  for i in range(n):
    for j in range(i+1, n):
      new.add(gcd(arr[i], arr[j]))
  # print(new)
 
  test = 1
 
  while True:
    if test not in new:
      print(test)
      break
    test+=1